crates/testing/src/sdk.rs

```diff
@@ -203,7 +208,93 @@ fn generate_bindings(language: &str, wasm_file: &str, client_project: &str, gene
             "--out-dir",
             generate_dir,
         ]);
-    })
+    });
+
+    let wasm_file = wasm_file.to_owned();
+
+    // Make sure that equivalent modules in all source languages result in the same ModuleDef.
+    //
+    // Unlike regular generate, this one is memoized by Wasm file instead of target directory,
+    // as we do want to rerun it per each module.
+    memoized!(|wasm_file: String| -> () {
+        let mut descriptions = spacetimedb_cli::generate::extract_descriptions(wasm_file.as_ref()).unwrap();
+
+        // Inline all typerefs in tables and reducers to make snapshots
+        // stable regardless of the order in which types are defined.
+
+        descriptions.inline_table_typerefs().unwrap();
+
+        let RawModuleDefV8 {
+            mut typespace,
+            mut tables,
+            mut reducers,
+            misc_exports,
+        } = descriptions;
+
+        for reducer in &mut reducers {
+            for arg in &mut reducer.args {
+                typespace.inline_typerefs_in_type(&mut arg.algebraic_type).unwrap();
+            }
+        }
+
+        for table in &mut tables {
+            // Rust generates same number of constraints as number of columns,
```

Comment:
@kazimuth Is this something that will be gone in the new schema by any chance?

Reply:
It is indeed! We still generate indexes for unique constraints, but all the other generated stuff is gone.